### PR TITLE
Add group profile methods and people 'remove' method

### PR DESCRIPTION
**Pull request by YayC**
See https://developer.mixpanel.com/docs/http#section-group-analytics

### CODE REVIEW

**Comment by YayC** [comment]
Looks like tests failed for node 6.9 since I'm using the spread operator (only in tests). @tdumitrescu should we still support that version? If not I can update .travis.yml to remove 6.9, otherwise I'll refactor that spread operator out.

**Comment by tdumitrescu** [comment]
Yeah, Node 6 is past EOL, we can remove it and use Object Spread.

File: test/profiles.js [hunk 1]
```diff
@@ -2,40 +2,53 @@ var Mixpanel    = require('../lib/mixpanel-node'),
...
+        if (['increment', 'append'].includes(func)) {
+            return test.done()
+        }
+    } else {
+        throw `Invalid namespace: ${this.namespace}`;
     }
-    var args = [this.distinct_id].concat(options.args || []);
 
-    if (options.use_modifiers) {
+    expected_data = {...expected_data, ...expected}
```
Comment (tdumitrescu) [inline test/profiles.js]:
```suggestion
    Object.assign(expected_data, expected);
```

File: test/profiles.js [hunk 2]
```diff
@@ -2,40 +2,53 @@ var Mixpanel    = require('../lib/mixpanel-node'),
...
-var test_send_request_args = function(test, func, options) {
-    var expected_data = {
-            $token: this.token,
-            $distinct_id: this.distinct_id,
-        };
-    for (var k in options.expected) {
-        expected_data[k] = options.expected[k];
+test_send_request_args = function(test, func, {args, expected, use_modifiers, use_callback} = {}) {
+    var endpoint;
+    var arguments;
```
Comment (tdumitrescu) [inline test/profiles.js]: by making a var `arguments` you're shadowing a reserved word that normally references the current function's arguments, so you'll want to choose a different identifier.

File: test/profiles.js [hunk 3]
```diff
@@ -615,3 +676,361 @@ exports.people = {
         },
     },
 };
+
+exports.groups = {
```
Comment (tdumitrescu) [inline test/profiles.js]: make separate test module

**Comment by YayC** [comment]
Thanks for your review @tdumitrescu! I made the suggested changes to DRY things out and pull out a groups module, but it required a more extensive refactor than anticipated. Do you think this looks reasonable?

**Comment by tdumitrescu** [comment]
superseded by https://github.com/mixpanel/mixpanel-node/pull/158

**Pull request closed without merging**